### DISCUSSION TITLE
Add tags to lists and add these tags automatically to new bookmarks/notes in that list - and a few other small ideas

### DISCUSSION THREAD

**Discussion by phil9309** [karakeep-app/karakeep#257] [Ideas]
Hi there,

first of all, Hoarder is great! I love it and use it daily. Thank you!

So, I checked the discussions here, I couldn't find this or something similar.

Is it possible to add a feature where we can add tags to lists, so if I add a new bookmark or note to this list, the bookmark/note will automatically get these tags attached too?

Lists and tags are kind of redundant in a certain way, so I think that would make the tagging flow a little easier.

What do you guys think?

Edit:

I thought of a few other things: 

Maybe an option to auto add the items in nested lists to their parent list? I collect all my notes in a "notes" list and then I have nested lists for different topics.

In the "all tags" overview, would it be possible to add a filter for items with no tags at all? So i can quickly see where I forgot to add a tag? This would be a feature similiar to the "unused tag" feature - just the other way around I guess.

**Comment by MohamedBassem** [comment]
Hi, I'm sorry I missed that thread.

> So if I add a new bookmark or note to this list, the bookmark/note will automatically get these tags attached too?

What we're planning to do I think can solve your problem in a different way. We're planning to introduced "smart lists". As in, you can build a list from a "saved search". So you can have an automatic list for all the items containing a certain set of tags. Would that make your life easier?

> Maybe an option to auto add the items in nested lists to their parent list?

I think this is also something were a powerful search can solve without explicitly adding all items to a list. Something like: "all items under a certain parent recursively".

> In the "all tags" overview, would it be possible to add a filter for items with no tags at all?

Do you use AI tagging? Because all those who use AI tagging (which I'd have assumed that this is most people) never have a bookmark without any tags :D

**Reply by flavioveloso3** [reply]
Hello!

Thank you so much for hoarder. It's the best bookmark manager I've ever seen!

Sorry to jump into into the conversation, but I have the same "issue." 
I would like all my bookmarks to be automatically added to all parent lists.
So, your answer caught my attention.
I didn't understand if it's currently possible to find all nested items using the search function? 
If so, could you please explain how to do that?

I think the planned "smart lists" might indirectly solve this problem. But wouldn't that require creating additional lists? For example, I have a list `Tools>Code>Repos`. To display all bookmarks in the parent `Tools`, I would have to create additional lists like `(Tools(Search)` and `Code(Search)` to show all bookmarks nested in `Tools` and `Tools->Code` respectively. Or did I misunderstand?

But if it's possible to modify the main `Tools` _(and Tools->Code)_ lists to display the results of the search query you mentioned, that would be fantastic and would resolve this issue.
